src/m365/spo/commands/listitem/listitem-roleassignment-add.ts

```diff
@@ -0,0 +1,271 @@
+      if (args.options.listId) {
+        requestUrl += `lists(guid'${formatting.encodeQueryParameter(args.options.listId)}')/`;
+      }
+      else if (args.options.listTitle) {
+        requestUrl += `lists/getByTitle('${formatting.encodeQueryParameter(args.options.listTitle)}')/`;
+      }
+      else if (args.options.listUrl) {
+        const listServerRelativeUrl: string = urlUtil.getServerRelativePath(args.options.webUrl, args.options.listUrl);
+        requestUrl += `GetList('${formatting.encodeQueryParameter(listServerRelativeUrl)}')/`;
+      }
+
+      requestUrl += `items(${args.options.listItemId})/`;
+
+      args.options.roleDefinitionId = await this.GetRoleDefinitionId(args.options);
+      if (args.options.upn) {
+        args.options.principalId = await this.GetUserPrincipalId(args.options);
+        await this.AddRoleAssignment(requestUrl, logger, args.options);
+      }
+      else if (args.options.groupName) {
+        args.options.principalId = await this.GetGroupPrincipalId(args.options);
+        await this.AddRoleAssignment(requestUrl, logger, args.options);
+      }
+      else {
+        await this.AddRoleAssignment(requestUrl, logger, args.options);
+      }
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  }
+
+  private AddRoleAssignment(requestUrl: string, logger: Logger, options: Options): Promise<void> {
+    const requestOptions: any = {
+      url: `${requestUrl}roleassignments/addroleassignment(principalid='${options.principalId}',roledefid='${options.roleDefinitionId}')`,
+      method: 'POST',
+      headers: {
+        'accept': 'application/json;odata=nometadata',
+        'content-type': 'application/json'
+      },
+      responseType: 'json'
+    };
+
+    return request
+      .post(requestOptions)
+      .then(_ => Promise.resolve())
+      .catch((err: any) => Promise.reject(err));
+  }
+
+  private GetRoleDefinitionId(options: Options): Promise<number> {
+    if (!options.roleDefinitionName) {
+      return Promise.resolve(options.roleDefinitionId as number);
+    }
+
+    const roleDefinitionListCommandOptions: SpoRoleDefinitionListCommandOptions = {
+      webUrl: options.webUrl,
+      output: 'json',
+      debug: this.debug,
+      verbose: this.verbose
+    };
+
+    return Cli.executeCommandWithOutput(SpoRoleDefinitionListCommand as Command, { options: { ...roleDefinitionListCommandOptions, _: [] } })
+      .then((output: CommandOutput): Promise<number> => {
+        const getRoleDefinitionListOutput = JSON.parse(output.stdout);
+        const roleDefinitionId: number = getRoleDefinitionListOutput.find((role: RoleDefinition) => role.Name === options.roleDefinitionName).Id;
+        return Promise.resolve(roleDefinitionId);
+      }, (err: CommandErrorWithOutput) => {
+        return Promise.reject(err);
+      });
+  }
+
+  private GetGroupPrincipalId(options: Options): Promise<number> {
```

Comment:
Let's use lowerCamelCase as naming convention for our methodes. `getGroupPrincipalId`